### DISCUSSION TITLE
Trying to navigate to new page that shows detailed data of the clicked item based on Id, but it always shows not found or error

### DISCUSSION THREAD

**Discussion by Alayers2804** [vercel/next.js#59929] [App Router]
https://github.com/Alayers2804/restaurant-sekawan.git

i am trying to navigate from main page to detail page, but it always shows error or not found. Does anyone know what to do? is there anything wrong with the code especially the hooks and the detail page inside the detail folder? thank you

**Comment by powercording** [comment]
~you are importing route from "next/navigation"~
~router form next navigation doesn't have query property.~

~try import useRrouter from "next/rotuer"~


useRouter from 'next/router' is not able in app dir.

try usePathname hook and extract id from string.
```jsx
// app/detail/[id]/page.jsx

"use client";
import React from "react";
 // import { useRouter } from "next/navigation"; ❌
import { usePathname } from "next/navigation" // try this. 

const page = () => {
  // const router = useRouter();
  const pathName = usePathName(); // try this. 

   // const { id } = router.query;
  const id = pathName.split("/").at(-1); // try this. 
  console.log(id);
  return <h1>{`Ini adalah artikel ${id}`}</h1>;
};

export default page;
```

**Reply by powercording** [reply]
If you are using 'useClient' for the 'useRouter' hook, and if there's no need for other client-side features, you can change 'detail/[id]/page' to a server component. This way, you can obtain the pathname without using any hooks.

You can find about [slug](https://nextjs.org/docs/app/building-your-application/routing/dynamic-routes) and get more information of dynamic route
